### PR TITLE
[DEV-749] Optimize the node_commit indexes for the new releases matching

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
@dennwc 
```
2020/08/25 14:15:20 99_footer.sql
unexpected hash: "ac74c6c02c8df3c72635f484427e03a3d770610c"; you must write a migration instead
exit status 1
sql/generate.go:5: running "go": exit status 1
Makefile:38: recipe for target 'gen-sql' failed
make: *** [gen-sql] Error 1
```
So must I write a migration as asked? How do I do that?

**Comment by vmarkovtsev** [comment]
@dennwc done

File: sql/migrations/003_optimize_node_commit_indexes.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DROP INDEX IF EXISTS github_node_commit_sha_cover;
```
Comment (dennwc) [inline sql/migrations/003_optimize_node_commit_indexes.up.sql:1]: Can you please keep this specific one? It's quite useful to search commits by SHA.

File: sql/migrations/003_optimize_node_commit_indexes.up.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+DROP INDEX IF EXISTS github_node_commit_pushed_date;
+DROP INDEX IF EXISTS github_node_commit_authored_date;
+DROP INDEX IF EXISTS github_node_commit_author;
+
+CREATE INDEX IF NOT EXISTS github_node_commit_release_match ON public.github_node_commit ("oid", "committed_date");
```
Comment (dennwc) [inline sql/migrations/003_optimize_node_commit_indexes.up.sql:6]: Do you filter by both fields at the same time?

If so, you either want:
```
CREATE INDEX IF NOT EXISTS github_node_commit_release_match ON public.github_node_commit ("oid") INCLUDE ("committed_date");
```
(filtering on `oid` almost always produces a single result)

or
```
CREATE INDEX IF NOT EXISTS github_node_commit_release_match ON public.github_node_commit ("committed_date","oid");
```
(so the commits are first ordered by the date, which gives multiple results and then by `oid`)

File: sql/migrations/003_optimize_node_commit_indexes.up.sql (hunk 1 above)
Comment (vmarkovtsev) [reply sql/migrations/003_optimize_node_commit_indexes.up.sql:1]: Please look around the file, we have the other index which allows to do the same.

File: sql/migrations/003_optimize_node_commit_indexes.up.sql (hunk 2 above)
Comment (vmarkovtsev) [reply sql/migrations/003_optimize_node_commit_indexes.up.sql:6]: The error in your logic is that I filter by one sha. I filter by hundreds of thousands of SHAs and then `committed_date` reduces the number of rows 20x. I ran performance tests and this is the best index for my query.

Comment (vmarkovtsev) [reply sql/migrations/003_optimize_node_commit_indexes.up.sql:6]: https://explain.tensor.ru/archive/explain/8e379d39eccf113387f80d4f0e806dbc:0:2020-08-26 vs https://explain.tensor.ru/archive/explain/f480047f537671249092ebebfd4684f8:0:2020-08-26

**Review by dennwc — APPROVED** [review]
Thanks!

**Pull request merged**